datasets/mwsc/mwsc.py

```diff
@@ -0,0 +1,135 @@
+"""A modification of the Winograd Schema Challenge to ensure answers are a single context word"""
+from __future__ import absolute_import, division, print_function
+
+import os
+import re
+
+import nlp
+
+
+_CITATION = """\
+@article{McCann2018decaNLP,
+  title={The Natural Language Decathlon: Multitask Learning as Question Answering},
+  author={Bryan McCann and Nitish Shirish Keskar and Caiming Xiong and Richard Socher},
+  journal={arXiv preprint arXiv:1806.08730},
+  year={2018}
+}
+"""
+
+_DESCRIPTION = """\
+Examples taken from the Winograd Schema Challenge modified to ensure that answers are a single word from the context.
+This modified Winograd Schema Challenge (MWSC) ensures that scores are neither inflated nor deflated by oddities in phrasing.
+"""
+
+_DATA_URL = "https://raw.githubusercontent.com/salesforce/decaNLP/1e9605f246b9e05199b28bde2a2093bc49feeeaa/local_data/schema.txt"
+# Alternate: https://s3.amazonaws.com/research.metamind.io/decaNLP/data/schema.txt
+
+class MWSC(nlp.GeneratorBasedBuilder):
+    """MWSC: modified Winograd Schema Challenge"""
+
+    VERSION = nlp.Version("0.1.0")
+
+    def _info(self):
+        return nlp.DatasetInfo(
+            description=_DESCRIPTION,
+            features=nlp.Features(
+                {
+                    "sentence": nlp.Value("string"),
+                    "question": nlp.Value("string"),
+                    "options": nlp.features.Sequence(nlp.Value("string")),
+                    "answer": nlp.Value("string"),
+                }
+            ),
+            # If there's a common (input, target) tuple from the features,
+            # specify them here. They'll be used if as_supervised=True in
+            # builder.as_dataset.
+            supervised_keys=None,
+            # Homepage of the dataset for documentation
+            homepage="http://decanlp.com",
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        """Returns SplitGenerators."""
+        schemas_file = dl_manager.download_and_extract(_DATA_URL)
+
+        if os.path.isdir(schemas_file):
+            # During testing the download manager mock gives us a directory
+            schemas_file = os.path.join(schemas_file,"schema.txt")
```

Comment:
Do you think there is a way to avoid this trick @patrickvonplaten ?

Reply:
I borrowed it from https://github.com/huggingface/nlp/blob/7d1526dfeeb29248d832f1073192dbf03ad642da/datasets/tiny_shakespeare/tiny_shakespeare.py#L76 so if there is we can fix it there too.

Reply:
Thanks. We'll keep that in mind for when we re-do a pass over dummy data I think.